components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2839,12 +2840,25 @@ private void readAndSetConfigurationsFromProperties(List<ServiceProviderProperty
 
                 if (USE_DOMAIN_IN_ROLES.equals(name)) {
                     localAndOutboundConfig.setUseUserstoreDomainInRoles(value == null || Boolean.parseBoolean(value));
+                    removableProperties.add(serviceProviderProperty);
```

Comment:
why do we have to add it to a list and remove it later? can't we remove it here itself?